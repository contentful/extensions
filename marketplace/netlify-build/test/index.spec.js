/* global global */
import React from 'react';
import { render, cleanup, wait, fireEvent } from '@testing-library/react';
import NetlifyExtension from '../src/index.js';

const mockaddListener = jest.fn();
const mocksubscribe = jest.fn();
const mockpublish = jest.fn();
const mockhistory = jest.fn(() =>
  Promise.resolve({
    messages: [
      {
        timetoken: '15620690994438613',
        entry: {
          contentful: true,
          event: 'triggered',
          userId: '2jvc3kU4n7OIABiFMTaGyB'
        }
      }
    ],
    startTimeToken: '15620690994438613',
    endTimeToken: '15620690994438613'
  })
);

jest.mock('pubnub', () => {
  return {
    __esModule: true,
    default: jest.fn(() => ({
      addListener: mockaddListener,
      subscribe: mocksubscribe,
      publish: mockpublish,
      history: mockhistory,
      removeListener: jest.fn(),
      unsubscribe: jest.fn()
    }))
  };
});

// date-fns provides functions such as `distanceInWordsToNow` which will change as the time changes
// it is necessary to ensure the date will remain the same no matter when these tests are run
global.Date.now = jest.fn(() => 1562082624100);

const sdkMock = {
  parameters: {
    instance: {},
    installation: {
      names: 'upbeat,friendly',
      channels:
        'c-wj2QelNg1OvkkWDF-5133ac93-7fc0-463b-96d0-bb799e7fcb07,c-wj2QelNg1OvkkWDF-119493e8-f2a3-4771-8b3c-f402b5966e84',
      publishKey: 'pub-c-a99421b9-4f21-467b-ac0c-d0292824e8e1',
      subscribeKey: 'sub-c-3992b1ae-f7c9-11e8-adf7-5a5b31762c0f',
      buildHookUrls:
        'https://api.netlify.com/build_hooks/5d19c2c1b359da07eca189af,https://api.netlify.com/build_hooks/5d19c2c1b359da2f7da18998',
      netlifySiteIds: '5133ac93-7fc0-463b-96d0-bb799e7fcb07,119493e8-f2a3-4771-8b3c-f402b5966e84',
      netlifySiteUrls:
        'https://upbeat-kare-a02208.netlify.com,https://friendly-easley-6bcf4a.netlify.com'
    }
  },
  window: {
    startAutoResizer: jest.fn()
  },
  space: {
    getUsers: jest.fn(() =>
      Promise.resolve({
        items: [
          {
            sys: {
              type: 'User',
              id: '2jvc3kU4n7OIABiFMTaGyB'
            },
            firstName: 'John',
            lastName: 'Smith',
            email: 'email@email.com',
            avatarUrl: 'something.com'
          }
        ]
      })
    )
  },
  user: {
    sys: {
      id: '2jvc3kU4n7OIABiFMTaGyB'
    }
  }
};

describe('NetlifyExtension', () => {
  afterEach(() => {
    cleanup();
    mockaddListener.mockClear();
    mocksubscribe.mockClear();
    mockpublish.mockClear();
    mockhistory.mockClear();
  });

  it('should render the NetlifySideBarBuildButton', async () => {
    const { container } = render(<NetlifyExtension sdk={sdkMock} />);

    expect(container).toMatchSnapshot();
  });

  it('should open the preview window with correct URL', async () => {
    const { getByTestId } = render(<NetlifyExtension sdk={sdkMock} />);

    expect(getByTestId('preview-button').href).toEqual('https://upbeat-kare-a02208.netlify.com/');

    // select second preview url
    fireEvent.change(getByTestId('site-selector'), { target: { value: '1' } });

    expect(getByTestId('preview-button').href).toEqual(
      'https://friendly-easley-6bcf4a.netlify.com/'
    );

    // should ensure some basic security practices
    expect(getByTestId('preview-button').rel).toEqual('noopener noreferrer');

    // should open in new window
    expect(getByTestId('preview-button').target).toEqual('_blank');
  });

  it('should subscribe to the correct channels', async () => {
    render(<NetlifyExtension sdk={sdkMock} />);

    await wait();
    expect(mocksubscribe).toHaveBeenCalledWith({ channels: ['5d19c2c1b359da07eca189af'] });
    expect(mockhistory).toHaveBeenCalledWith({
      channel: '5d19c2c1b359da07eca189af',
      count: 25,
      stringifiedTimeToken: true
    });
  });

  it('should show a triggering message when a build is triggered with username', async () => {
    const { container } = render(<NetlifyExtension sdk={sdkMock} />);

    await wait();

    expect(container).toMatchSnapshot();
  });

  it('should show a triggering button and build message without username if not loaded yet', async () => {
    sdkMock.space.getUsers.mockResolvedValue(Promise.resolve({ items: [] }));

    const { container } = render(<NetlifyExtension sdk={sdkMock} />);

    await wait();

    expect(container).toMatchSnapshot();
  });

  it('should should show a building message when a build is building', async () => {
    mockhistory.mockReturnValue(
      Promise.resolve({
        messages: [
          {
            timetoken: '15620690994438613',
            entry: {
              contentful: true,
              event: 'triggered',
              userId: '2jvc3kU4n7OIABiFMTaGyB'
            }
          },
          {
            timetoken: '15620690994438653',
            entry: {
              contentful: true,
              event: 'build-started',
              userId: '2jvc3kU4n7OIABiFMTaGyB'
            }
          }
        ],
        startTimeToken: '15620690994438613',
        endTimeToken: '15620690994438653'
      })
    );

    const { container } = render(<NetlifyExtension sdk={sdkMock} />);

    await wait();
    expect(container).toMatchSnapshot();
  });

  it('should should show a clickable build button when build is ready', async () => {
    mockhistory.mockReturnValue(
      Promise.resolve({
        messages: [
          {
            timetoken: '15620690994438613',
            entry: {
              contentful: true,
              event: 'triggered',
              userId: '2jvc3kU4n7OIABiFMTaGyB'
            }
          },
          {
            timetoken: '15620690994438653',
            entry: {
              contentful: true,
              event: 'build-started',
              userId: '2jvc3kU4n7OIABiFMTaGyB'
            }
          },
          {
            timetoken: '15620690994438693',
            entry: {
              contentful: true,
              event: 'build-ready',
              userId: '2jvc3kU4n7OIABiFMTaGyB'
            }
          }
        ],
        startTimeToken: '15620690994438613',
        endTimeToken: '15620690994438693'
      })
    );

    const { container } = render(<NetlifyExtension sdk={sdkMock} />);

    await wait();
    expect(container).toMatchSnapshot();
  });
});
