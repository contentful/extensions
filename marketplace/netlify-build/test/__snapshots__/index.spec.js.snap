// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetlifyExtension should render the NetlifySideBarBuildButton 1`] = `
<div>
  <div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--disabled___1E20M Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      data-testid="build-button"
      disabled=""
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Build
        </span>
      </span>
    </button>
  </div>
  <a
    class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1f4d741 Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
    data-test-id="cf-ui-button"
    data-testid="preview-button"
    href="https://upbeat-kare-a02208.netlify.com"
    rel="noopener noreferrer"
    target="_blank"
    type="button"
  >
    <span
      class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
      tabindex="-1"
    >
      <span
        class="Button__Button__label___3tcOj"
      >
        <div
          class="css-1dnzbhn"
        >
          <svg
            class="Icon__Icon___38Epv css-1qv3o9n Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Open preview
        </div>
      </span>
    </span>
  </a>
  <div
    class="Select__Select__wrapper___2pBIz Select__Select--full___7ei7d"
  >
    <select
      class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
      data-test-id="cf-ui-select"
      data-testid="site-selector"
    >
       
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-0"
        value="0"
      >
        upbeat
      </option>
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-1"
        value="1"
      >
        friendly
      </option>
    </select>
    <svg
      class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;

exports[`NetlifyExtension should should show a building message when a build is building 1`] = `
<div>
  <div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--disabled___1E20M Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      data-testid="build-button"
      disabled=""
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Triggering...
        </span>
        <svg
          class="Spinner__Spinner___32lpa Button__Button__spinner___3j8Aj Spinner__Spinner--small___2hyo0 Spinner__Spinner--white___3Hsq3 Button__Button--spinner--enter___1qgg7 Button__Button--spinner-active___EEKjQ"
          data-test-id="cf-ui-spinner"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Loading…
          </title>
          <path
            d="M2,10a8,8,0,0,1,8-8V0a10,10,0,0,0,0,20V18A8,8,0,0,1,2,10Z"
          />
          <path
            d="M10,0V2a8,8,0,0,1,0,16v2A10,10,0,0,0,10,0Z"
            style="opacity: 0.4;"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-fvpxna"
    >
      Triggered about 4 hours ago at 12:04:59 PM.
    </div>
  </div>
  <a
    class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1f4d741 Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
    data-test-id="cf-ui-button"
    data-testid="preview-button"
    href="https://upbeat-kare-a02208.netlify.com"
    rel="noopener noreferrer"
    target="_blank"
    type="button"
  >
    <span
      class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
      tabindex="-1"
    >
      <span
        class="Button__Button__label___3tcOj"
      >
        <div
          class="css-1dnzbhn"
        >
          <svg
            class="Icon__Icon___38Epv css-1qv3o9n Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Open preview
        </div>
      </span>
    </span>
  </a>
  <div
    class="Select__Select__wrapper___2pBIz Select__Select--full___7ei7d"
  >
    <select
      class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
      data-test-id="cf-ui-select"
      data-testid="site-selector"
    >
       
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-0"
        value="0"
      >
        upbeat
      </option>
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-1"
        value="1"
      >
        friendly
      </option>
    </select>
    <svg
      class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;

exports[`NetlifyExtension should should show a clickable build button when build is ready 1`] = `
<div>
  <div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--disabled___1E20M Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      data-testid="build-button"
      disabled=""
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Triggering...
        </span>
        <svg
          class="Spinner__Spinner___32lpa Button__Button__spinner___3j8Aj Spinner__Spinner--small___2hyo0 Spinner__Spinner--white___3Hsq3 Button__Button--spinner--enter___1qgg7 Button__Button--spinner-active___EEKjQ"
          data-test-id="cf-ui-spinner"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Loading…
          </title>
          <path
            d="M2,10a8,8,0,0,1,8-8V0a10,10,0,0,0,0,20V18A8,8,0,0,1,2,10Z"
          />
          <path
            d="M10,0V2a8,8,0,0,1,0,16v2A10,10,0,0,0,10,0Z"
            style="opacity: 0.4;"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-fvpxna"
    >
      Triggered about 4 hours ago at 12:04:59 PM.
    </div>
  </div>
  <a
    class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1f4d741 Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
    data-test-id="cf-ui-button"
    data-testid="preview-button"
    href="https://upbeat-kare-a02208.netlify.com"
    rel="noopener noreferrer"
    target="_blank"
    type="button"
  >
    <span
      class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
      tabindex="-1"
    >
      <span
        class="Button__Button__label___3tcOj"
      >
        <div
          class="css-1dnzbhn"
        >
          <svg
            class="Icon__Icon___38Epv css-1qv3o9n Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Open preview
        </div>
      </span>
    </span>
  </a>
  <div
    class="Select__Select__wrapper___2pBIz Select__Select--full___7ei7d"
  >
    <select
      class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
      data-test-id="cf-ui-select"
      data-testid="site-selector"
    >
       
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-0"
        value="0"
      >
        upbeat
      </option>
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-1"
        value="1"
      >
        friendly
      </option>
    </select>
    <svg
      class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;

exports[`NetlifyExtension should show a triggering button and build message without username if not loaded yet 1`] = `
<div>
  <div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--disabled___1E20M Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      data-testid="build-button"
      disabled=""
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Triggering...
        </span>
        <svg
          class="Spinner__Spinner___32lpa Button__Button__spinner___3j8Aj Spinner__Spinner--small___2hyo0 Spinner__Spinner--white___3Hsq3 Button__Button--spinner--enter___1qgg7 Button__Button--spinner-active___EEKjQ"
          data-test-id="cf-ui-spinner"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Loading…
          </title>
          <path
            d="M2,10a8,8,0,0,1,8-8V0a10,10,0,0,0,0,20V18A8,8,0,0,1,2,10Z"
          />
          <path
            d="M10,0V2a8,8,0,0,1,0,16v2A10,10,0,0,0,10,0Z"
            style="opacity: 0.4;"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-fvpxna"
    >
      Triggered about 4 hours ago at 12:04:59 PM.
    </div>
  </div>
  <a
    class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1f4d741 Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
    data-test-id="cf-ui-button"
    data-testid="preview-button"
    href="https://upbeat-kare-a02208.netlify.com"
    rel="noopener noreferrer"
    target="_blank"
    type="button"
  >
    <span
      class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
      tabindex="-1"
    >
      <span
        class="Button__Button__label___3tcOj"
      >
        <div
          class="css-1dnzbhn"
        >
          <svg
            class="Icon__Icon___38Epv css-1qv3o9n Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Open preview
        </div>
      </span>
    </span>
  </a>
  <div
    class="Select__Select__wrapper___2pBIz Select__Select--full___7ei7d"
  >
    <select
      class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
      data-test-id="cf-ui-select"
      data-testid="site-selector"
    >
       
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-0"
        value="0"
      >
        upbeat
      </option>
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-1"
        value="1"
      >
        friendly
      </option>
    </select>
    <svg
      class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;

exports[`NetlifyExtension should show a triggering message when a build is triggered with username 1`] = `
<div>
  <div>
    <button
      class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-82qlwu Button__Button--primary___JImeO Button__Button--disabled___1E20M Button__Button--full-width___3Fmpo"
      data-test-id="cf-ui-button"
      data-testid="build-button"
      disabled=""
      type="button"
    >
      <span
        class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
        tabindex="-1"
      >
        <span
          class="Button__Button__label___3tcOj"
        >
          Triggering...
        </span>
        <svg
          class="Spinner__Spinner___32lpa Button__Button__spinner___3j8Aj Spinner__Spinner--small___2hyo0 Spinner__Spinner--white___3Hsq3 Button__Button--spinner--enter___1qgg7 Button__Button--spinner-active___EEKjQ"
          data-test-id="cf-ui-spinner"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <title>
            Loading…
          </title>
          <path
            d="M2,10a8,8,0,0,1,8-8V0a10,10,0,0,0,0,20V18A8,8,0,0,1,2,10Z"
          />
          <path
            d="M10,0V2a8,8,0,0,1,0,16v2A10,10,0,0,0,10,0Z"
            style="opacity: 0.4;"
          />
        </svg>
      </span>
    </button>
    <div
      class="css-fvpxna"
    >
      Triggered by John Smith about 4 hours ago at 12:04:59 PM.
    </div>
  </div>
  <a
    class="Button__Button___1ZfFj a11y__focus-border--default___60AXp css-1f4d741 Button__Button--muted___2Wair Button__Button--full-width___3Fmpo"
    data-test-id="cf-ui-button"
    data-testid="preview-button"
    href="https://upbeat-kare-a02208.netlify.com"
    rel="noopener noreferrer"
    target="_blank"
    type="button"
  >
    <span
      class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
      tabindex="-1"
    >
      <span
        class="Button__Button__label___3tcOj"
      >
        <div
          class="css-1dnzbhn"
        >
          <svg
            class="Icon__Icon___38Epv css-1qv3o9n Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 19H5V5h7V3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          Open preview
        </div>
      </span>
    </span>
  </a>
  <div
    class="Select__Select__wrapper___2pBIz Select__Select--full___7ei7d"
  >
    <select
      class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
      data-test-id="cf-ui-select"
      data-testid="site-selector"
    >
       
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-0"
        value="0"
      >
        upbeat
      </option>
      <option
        data-test-id="cf-ui-select-option"
        data-testid="option-1"
        value="1"
      >
        friendly
      </option>
    </select>
    <svg
      class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;
